grovedb/src/replication.rs

```diff
@@ -1,268 +1,58 @@
-                                prefix,
-                                (current_path.to_vec(), actual_value_hash, elem_value_hash),
-                            );
-                        }
-                    }
-                    Some(t) => {
-                        let parent_merk = self
-                            .open_transactional_merk_at_path(parent_path, t, None, grove_version)
-                            .value?;
-                        if let Ok(Some((elem_value, elem_value_hash))) = parent_merk
-                            .get_value_and_value_hash(
-                                parent_key,
-                                true,
-                                None::<&fn(&[u8], &GroveVersion) -> Option<ValueDefinedCostType>>,
-                                grove_version,
-                            )
-                            .value
-                        {
-                            let actual_value_hash = value_hash(&elem_value).unwrap();
-                            subtrees_metadata.data.insert(
-                                prefix,
-                                (current_path.to_vec(), actual_value_hash, elem_value_hash),
-                            );
-                        }
-                    }
-                },
-                _ => {
-                    subtrees_metadata.data.insert(
-                        prefix,
-                        (
-                            current_path.to_vec(),
-                            CryptoHash::default(),
-                            CryptoHash::default(),
-                        ),
-                    );
-                }
-            }
-        }
-        Ok(subtrees_metadata)
-    }
 
     // Fetch a chunk by global chunk id (should be called by ABCI when
     // LoadSnapshotChunk method is called) Params:
     // global_chunk_id: Global chunk id in the following format:
-    // [SUBTREE_PREFIX:CHUNK_ID] SUBTREE_PREFIX: 32 bytes (mandatory) (All zeros
-    // = Root subtree) CHUNK_ID: 0.. bytes (optional) Traversal instructions to
+    // [SUBTREE_PREFIX:SIZE_ROOT_KEY:ROOT_KEY:IS_SUM_TREE:CHUNK_ID]
+    // SUBTREE_PREFIX: 32 bytes (mandatory) (All zeros = Root subtree)
+    // SIZE_ROOT_KEY: 1 byte: Size of ROOT_KEY in bytes
+    // ROOT_KEY: SIZE_ROOT_KEY bytes (optional)
+    // IS_SUM_TREE: 1 byte (mandatory) marks if the tree is a sum tree or not
+    // CHUNK_ID: 0.. bytes (optional) Traversal instructions to
     // the root of the given chunk. Traversal instructions are "1" for left, and
     // "0" for right. TODO: Compact CHUNK_ID into bitset for size optimization
     // as a subtree can be big hence traversal instructions for the deepest chunks
-    // tx: Transaction. Function returns the data by opening merks at given tx.
+    // transaction: Transaction. Function returns the data by opening merks at given
+    // tx. version: Version of state sync protocol version
+    // grove_version: Version of groveDB
```

Comment:
Doc comments start with `///` for the following item, while `//` are regular comments